C/plugins/north/http-north/plugin.cpp

```diff
@@ -0,0 +1,253 @@
+const string& getReadingString(const Reading& reading);
+
+/**
+ * Return the information about this plugin
+ */
+PLUGIN_INFORMATION *plugin_info()
+{
+	return &info;
+}
+
+/**
+ * Return default plugin configuration:
+ * plugin specific and types_id
+ */
+const map<const string, const string>& plugin_config()
+{
+	return plugin_configuration;
+}
+
+/**
+ * Initialise the plugin with configuration.
+ *
+ * This funcion is called to get the plugin handle.
+ */
+PLUGIN_HANDLE plugin_init(map<string, string>&& configData)
+{
+	Logger::getLogger()->info("http-north C plugin: %s", __FUNCTION__);
+
+	/**
+	 * Handle the HTTP(S) parameters here
+	 */
+	ConfigCategory configCategory("cfg", configData["GLOBAL_CONFIGURATION"]);
+	string url = configCategory.getValue("URL");
+	unsigned int timeout = atoi(configCategory.getValue("HttpTimeout").c_str());
+
+	/**
+	 * Extract host, port, path from URL
+	 */
+	size_t findProtocol = url.find_first_of(":");
+	string protocol = url.substr(0,findProtocol);
+
+	string tmpUrl = url.substr(findProtocol + 3);
+	size_t findPort = tmpUrl.find_first_of(":");
+	string hostName = tmpUrl.substr(0, findPort);
+
+	size_t findPath = tmpUrl.find_first_of("/");
+	string port = tmpUrl.substr(findPort + 1 , findPath - findPort -1);
+	string path = tmpUrl.substr(findPath);
+
+	/**
+	 * Allocate the HTTP(S) handler for "Hostname : port",
+	 * connect_timeout and request_timeout.
+	 */
+	string hostAndPort(hostName + ":" + port);	
+
+	CONNECTOR_INFO *connector_info = new CONNECTOR_INFO;
+	if (protocol == string("http"))
+		connector_info->sender = new SimpleHttp(hostAndPort, timeout, timeout);
+	else if (protocol == string("https"))
+		connector_info->sender = new SimpleHttps(hostAndPort, timeout, timeout);
+	else
+	{
+		Logger::getLogger()->error("Didn't find http/https prefix in URL='%s', cannot proceed", url.c_str());
+		throw new exception();
+	}
+
+	connector_info->path = path;
+	connector_info->proto = protocol;
+	Logger::getLogger()->info("HTTP plugin configured: URL='%s' ", url.c_str());
+
+	return (PLUGIN_HANDLE) (connector_info);
+}
+
+/**
+ * Send Readings data to historian server
+ */
+uint32_t plugin_send(const PLUGIN_HANDLE handle,
+		     const vector<Reading *> readings)
+{
+	CONNECTOR_INFO *connInfo = (CONNECTOR_INFO *) handle;
+	return sendToServer(readings, connInfo);
+}
+
+/**
+ * Shutdown the plugin
+ *
+ * Delete allocated data
+ *
+ * @param handle    The plugin handle
+ */
+void plugin_shutdown(PLUGIN_HANDLE handle)
+{
+	Logger::getLogger()->info("http-north C plugin: %s", __FUNCTION__);
+	CONNECTOR_INFO *connInfo = (CONNECTOR_INFO *) handle;
+	delete connInfo->sender;
+	delete connInfo;
+}
+
+const string& getReadingString(const Reading& reading)
+{
+	string *value = new string;
+	string &m_value = *value;
+
+	// Convert reading data into JSON string
+	m_value.append("{\"timestamp\" : \"" + reading.getAssetDateTime(Reading::FMT_STANDARD) + "Z" + "\"");
```

Comment:
if ingested readings has +05:30, then ?

Reply:
I saw the readings with same and correct UTC timestamps consistently at both http-north C (ubuntu VM) as well as http-south python (Raspberry Pi) ends. Is that correct behavior or something needs to change?